org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/dom/TypeSignatureHelper.java

```diff
@@ -0,0 +1,152 @@
+/*******************************************************************************
+ * Copyright (c) 2024, Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.jdt.internal.javac.dom;
+
+import org.eclipse.jdt.core.dom.ArrayType;
+import org.eclipse.jdt.core.dom.NameQualifiedType;
+import org.eclipse.jdt.core.dom.ParameterizedType;
+import org.eclipse.jdt.core.dom.PrimitiveType;
+import org.eclipse.jdt.core.dom.QualifiedType;
+import org.eclipse.jdt.core.dom.SimpleType;
+import org.eclipse.jdt.core.dom.Type;
+import org.eclipse.jdt.core.dom.WildcardType;
+
+/**
+ * Utility class to help convert {@link org.eclipse.jdt.core.dom.Type} into a string type signature.
+ *
+ * For the format of the strings, see {@link org.eclipse.jdt.core.Signature}.
+ */
+public class TypeSignatureHelper {
```

Comment:
Have you tried using `org.eclipse.jdt.internal.core.util.Util.getSignature(Type)`? I think we should use it as much as possible, and if necessary contribute patches upstream.

Reply:
Yes, I've tried it, that method doesn't support generics unfortunately. I guess I should add this into the `Signature` class and push it upstream.

Reply:
I see it handles a case for `ASTNode.PARAMETERIZED_TYPE` so I suspect it's supposed to support generics? In any case, if we can push such change upstream, it would indeed be best.

Reply:
Oops no I've only tried methods in `Signature`. I'll look into that

Reply:
It says that it doesn't support parameterized types, but it seems to work. Good catch!

Reply:
Cool. Feel free to submit a change to the documentation upstream by the way.